File: Java/src/count-char.java
import java.util.TreeMap;

class CountChar {
    public static void main(String[] args) {
        String inputString = "12345678900000000000-Hello-World";
        countChar(inputString);
    }

    public static void countChar(String tmpString) {
        // use TreeMap which is naturally sorted
        TreeMap<Character, Integer> counts = new TreeMap<>();

        for (char letter : tmpString.toCharArray()) {
            counts.put(letter, counts.getOrDefault(letter, 0) + 1);
        }

        for (char i : counts.keySet()) {
            double percent = counts.get(i).floatValue() / tmpString.length();
            System.out.printf("'%c' %d %.2f\n", i, counts.get(i), percent);
        }
    }
}

File: Kotlin/src/count-char.kt
import java.util.*

fun main() {
    val stringAsACharSequence = "12345678900000000000-Hello-World"
    countChar(stringAsACharSequence)
}

fun countChar(tmpString: String) {
    // use TreeMap which is naturally sorted
    val counts = TreeMap<Char, Int>()

    tmpString.forEach { letter ->
        counts[letter] = counts.getOrPut(letter, { 0 }) + 1
    }

    for ((letter, value) in counts) {
        val tmp = value.toDouble() / tmpString.count()
        println("'%c' %d %.2f".format(letter, value, tmp))
    }
}
